### PR TITLE
Reworked Column Configuration in the data section

### PR DESCRIPTION
**Pull request by PClmnt**
## Description
This PR addresses the work needed to move the column configuration part of the data section (both edit and creating columns) into popovers that expand out from the column headers themselves. It also adds a new component for adding options, allowing the user to drag and drop as well as select colours for each option.

Screenshots: 

image

image

- Some fields have been updated with new UX. 

image
image

- With regards to the new option component, all options will be shown with a grey colour as default.  Selecting a colour in the above popover saves it, and ensures that will be the colour shown in the dropdown within the grid, if a colour is not selected, it will work as it currently does, assigning colours up to a certain amount before starting again. 

The following colors are available in the popover:

```
    "hsla(0, 90%, 75%, 0.3)",
    "hsla(50, 80%, 75%, 0.3)",
    "hsla(120, 90%, 75%, 0.3)",
    "hsla(200, 90%, 75%, 0.3)",
    "hsla(240, 90%, 75%, 0.3)",
    "hsla(320, 90%, 75%, 0.3)",
```

### CODE REVIEW

**Review by aptkingston — CHANGES_REQUESTED** [review]
Nice work @PClmnt! I've pulled this down and gone over it with Andy as well.

First thing that needs addressed is that it's impossible to change any data if your table contains an options column - it complains about an invalid validator:
image

In terms of functionality this all seems to work really well. There were a few small issues I found:
- The invalid validator message I mentioned above when attempting to change row data (not when saving a column)
- When editing a column (not creating) we should restrict which types are available to change to. Currently you can pick any type to change to, and picking certain types breaks the popover and makes everything disabled. This is incorrect and there are only certain types you're allowed to swap between.
- For the option colours, it's impossible to tell if you manually selected a color or of its an automatic color. The difference matters, because when you reorder options the color will change if it's an automatic one. We think it would be a nice update if when editing options we showed all automatic colors as just grey (but in the real grid they'll still use the proper colors). By showing them as grey it's just us saying "you haven't picked a color yet, so we'll automatically assign one".
- For option colours again, could we include a blue color? The 6 available colours aren't the best palette, so it would be nice to at least include the blue color (color 0 I believe) instead of one of the green shades

There are a few design changes Andy picked up on which would be good to change, just to add more polish to this.
- Remove the `Type` label on the type field when creating/editing columns
- Reduce spacing in between the fields to 16px rather than 20px
- Add the `quiet` prop to the delete button so that it doesn't have a border
- Change copy of save button to "Save" rather than "Save column"
- Update the background colour of the whole popover to be the lighter background color rather than the darker one (can't remember if this is `--background` or `--background-alt`), but we think it would make it easier to distinguish the popover from the actual grid, and help with adding contrast between input fields

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte [hunk 1]
```diff
@@ -155,60 +164,84 @@
...
+<div>
+  <Popover
+    bind:open
+    bind:this={popover}
+    {anchor}
+    align="right"
+    offset={0}
+    popoverTarget={document.getElementById(`grid-${rand}`)}
+    animate={false}
+    customZindex={100}
```
[the remark this answers is not in the document]
Comment (PClmnt) [reply packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:176]: I needed to add this, as when opening a Modal (say the formula field) the new popover would sit above the modal covering it. This ensures it stays below it. I'm open to other options with regards to how to handle this though as I know it's a weird prop to be available on the Popover component.

[the remark this answers is not in the document]
Comment (aptkingston) [reply packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:176]: Ah, understood! This is probably the first example of a modal being opened from inside a popover (normally it's the other way round).

I would maybe just update the popover to actually use the value passed in for `customZindex` - I noticed it always sets a z-index of 100 regardless of what the passed in value is.

**Review by aptkingston — APPROVED** [review]
LGTM! All seems sorted - nice work @PClmnt.

**Pull request merged**